torch/_inductor/fx_passes/post_grad.py

```diff
@@ -1015,6 +1015,29 @@ def fused_int_mm_mul(match: Match, mat1, mat2, mat3, out_dtype=None):
     return inductor.kernel.mm.tuned_fused_int_mm_mul(mat1, mat2, mat3, out_dtype)
 
 
+def is_index_put_and_requires_h2d_sync_for_cuda_value(node):
+    if node.target is not torch.ops.aten.index_put.default:
+        return False
+    # Inductor falls back to aten.index_put_.
+    # index_put_ will will call nonzero() and perform a H2D sync if
+    # any of its indices are bool/byte tensors
+    # However, it will short-circuit this H2D sync and run mask_fill_
+    # if the value we are putting is a cpu scalar.
+    # Therefore, when inductor sees an index_put_ with byte tensor indices,
+    # it should *not* convert the cpu scalar value into a cuda tensor.
+    any_byte_bool_indices = False
+    indices = node.args[1]
```

Comment:
nit: use `operator_schemas.normalize_function` instead of directly accessing args